cmake/SeastarDependencies.cmake

```diff
@@ -103,6 +104,12 @@ macro (seastar_find_dependencies)
     ucontext
     yaml-cpp)
 
+  # Only expose dpdk dependency during build. Since dpdk is included to within seastar library,
+  # there is no need to force it to be discoverable via find_package().
+  if (args_BUILD)
+    list (APPEND _seastar_all_dependencies dpdk)
+  endif()
```

Comment:
i'd like to understand your use case better. do you enable `Seastar_DPDK` when building Seastar?

Reply:
I build and use Seastar using the following:

```
./3rdparty/seastar/configure.py --mode=release --enable-dpdk --prefix=$(pwd)/build-install
ninja -C ./3rdparty/seastar/build/release install

...

mkdir -p build
cmake -S . -B build -GNinja \
    -DCMAKE_BUILD_TYPE=Release \
    -DCMAKE_PREFIX_PATH=$(pwd)/build-install \
    -DCMAKE_INSTALL_PREFIX=$(pwd)/build-install
ninja -C build
```

I would say this is pretty unconventional setup except for the install prefix and that DPDK is enabled. Without this PR build will fail because there's no dpdk installed in build-install (which is expected)

Reply:
@p12tic how do you build DPDK? do you build a static build of DPDK using something like `cooking.sh` ? i cannot find `--cook dpdk` in your command line arguments passed to `configure.py` .

but in a more unconventional setup, user might want to build Seastar with a dynamic build of DPDK libraries. in that case, we would have to find dpdk when building seastar *and* when building the seastar application. in this case, we cannot skip the `find_package(dpdk)` call when building seastar application, i think.